realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -137,6 +137,10 @@
     private final Map<Class<? extends RealmObject>, Table> classToTable =
             new HashMap<Class<? extends RealmObject>, Table>();
 
+    // Reference count on currently opened Realm instances.
+    // We need to know if all Realm instance of all threads are closed in order to clean up validatedRealmFiles.
+    private static final Map<String, Integer> realmFileReferenceCounter = new HashMap<String, Integer>();
```

Comment:
Perhaps use the term "typedRealm" and "dynamicRealm" instead of "realm" and "dynamicRealm"?.....otherwise it gets really hard to think about the variables IMO